File: Examples/python/hello.py
# Hello World in Python (sandboxed)
# Demonstrates basic Python syntax inside Time Warp Studio

name = "World"
print(f"Hello, {name}!")

# Greet several people
people = ["Alice", "Bob", "Charlie"]
for person in people:
    print(f"Hello, {person}!")

# String operations
greeting = "hello, python"
print(greeting.upper())
print(greeting.title())
print(f"Length: {len(greeting)}")

# Numbers and arithmetic
x, y = 42, 7
print(f"{x} / {y} = {x / y:.2f}")
print(f"{x} // {y} = {x // y}")
print(f"{x} % {y} = {x % y}")
print(f"{x} ** 2 = {x ** 2}")

# Ranges and list comprehension
squares = [i ** 2 for i in range(1, 6)]
print("Squares:", squares)

# Dictionary
person = {"name": "Ada Lovelace", "born": 1815, "field": "Mathematics"}
for key, value in person.items():
    print(f"  {key}: {value}")

print("Done!")

File: Examples/ruby/hello.rb
# Hello World in Ruby
# Demonstrates basic output, string interpolation, and variables

name = "World"
puts "Hello, #{name}!"

# Greet several people
["Alice", "Bob", "Charlie"].each do |person|
  puts "Hello, #{person}!"
end

# String methods
greeting = "hello, ruby"
puts greeting.upcase
puts greeting.capitalize
puts greeting.length

# Numbers and arithmetic
x = 42
y = 7
puts "#{x} divided by #{y} is #{x / y}"
puts "#{x} modulo #{y} is #{x % y}"
puts "#{x} to the power of 2 is #{x ** 2}"

# Ranges
(1..5).each { |i| puts i }

puts "Done!"
